File: 0797-all-paths-from-source-to-target/0797-all-paths-from-source-to-target.cpp
class Solution {
private:
    int n;
    vector<vector<int>> ans, adj;
    vector<int> tmp;
    
public:
    void backTracking(vector<vector<int>>& graph, int u) {
        if (u == n - 1) {
            ans.push_back(tmp);
            return;
        }
        for (int v : graph[u]) {
            tmp.push_back(v);
            backTracking(graph, v);
            tmp.pop_back();
        }
    }
    vector<vector<int>> allPathsSourceTarget(vector<vector<int>>& graph) {
        this->n = graph.size();
        adj.resize(n);
        tmp.push_back(0);
        backTracking(graph, 0);
        return ans;
    }
};

File: 0797-all-paths-from-source-to-target/0797-all-paths-from-source-to-target.py
class Solution:
    def allPathsSourceTarget(self, graph: List[List[int]]) -> List[List[int]]:
        ans = []
        n = len(graph)

        def dfs(u: int, path: List[int]) -> None:
            if u == n - 1:
                ans.append(path)
                return
            for v in graph[u]:
                dfs(v, path + [v])

        dfs(0, [0])

        return ans